### PR TITLE
[shardformer] bloom support flash attention

### PR DESCRIPTION
**Pull request by flybird1111**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

#4184 

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

bloom support flash attention

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/shardformer/modeling/bloom.py [hunk 1]
```diff
@@ -67,3 +70,90 @@ def build_bloom_alibi_tensor(self, attention_mask: torch.Tensor, num_heads: int,
...
+        residual (`torch.tensor`, *required*):
+            esidual tensor
+        prob (`float`, *required*):
+            dropout probability
+        training (`bool`, *required*):
+            training mode
+    """
+    out = F.dropout(x, p=prob, training=training)
+    out = residual + out
+    return out
```
Comment (FrankLeeeee) [inline colossalai/shardformer/modeling/bloom.py:159]: There is a fused dropout and addition, you can find it in https://github.com/hpcaitech/ColossalAI/blob/main/colossalai/kernel/jit/bias_dropout_add.py

**Comment by flybird1111** [comment]
image
All shardformer tests are passed.

**Review by FrankLeeeee — APPROVED** [review]
